legion_test/legion_test/utils.py

```diff
@@ -0,0 +1,124 @@
+#
+#    Copyright 2018 EPAM Systems
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+import logging
+import threading
+import time
+import re
+
+LOGGER = logging.getLogger(__name__)
+
+
+def wait_until(condition, iteration_duration=5, iterations=10):
+    """
+    Wait until condition would be true
+
+    :param condition: callable condition that returns bool value
+    :type condition: Callable
+    :param iteration_duration: duration between checks in seconds
+    :type iteration_duration: int
+    :param iterations: maximum count of iterations
+    :type iterations: int
+    :return: True or False -- is condition result has been changed to True
+    """
+    for iteation in range(iterations):
+        if condition():
+            return True
+
+        time.sleep(iteration_duration)
+
+    return False
+
+
+def normalize_name(name, dns_1035=False):
```

Comment:
How is this different from normalize_name in legion_utils?

Reply:
Similar, but legion_template does not include legion

Reply:
* write about Grafana and pandas

Reply:
We cannot import `legion` in `legion_template` because legion has many dependencies (like `pandas` and `numpy`) and too big for installation. `legion_test` are using in Grafana container bootstrapping, so if we add dependency on `legion` it will enlarge Grafana image size and build time.

Reply:
Have come to the conclusion that this is left here